vtex/billing-app
# Prevent free apps from making contracts in Billing

#### What is the purpose of this pull request?
Check the _free app_ corner case: free apps should not create new contracts in VTEX Billing system

#### What problem is this solving?
When trying to use the `purchase` endpoint with a free app, a contract would be posted in Billing, getting a error in response. With this change, the purchase is confirmed without touching it.

#### How should this be manually tested?
1. Clone/pull this repo and link in your workspace
2. Open its GraphiQL client and try to purchase a free app, such as `visa-checkout` e.g.
```
mutation {
  purchase(appName: "vtex.visa-checkout", termsOfUseAccepted: true) {
    purchased
    billingOptions
  }
}
```

The response should now be
```
{
  "data": {
    "purchase": {
      "purchased": true,
      "billingOptions": "{\n  \"free\": true\n}"
    }
  },
  "extensions": {}
}
```

#### Screenshots or example usage

#### Types of changes
